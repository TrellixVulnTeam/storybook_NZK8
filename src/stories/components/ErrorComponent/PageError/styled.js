// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';
import sadIcon from 'assets/commonIcons/sad.svg';

export const AlertIcon = styled(sadIcon)`
  height: 60px;
  width: 60px;
  fill: ${colors.strawberry};
  }
`;

export const WrapperDiv = styled.div`
  display: flex;
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.1);
  border-radius: 6px;
  border: solid 1px ${colors.paleGrey};
  flex-flow: column;
  align-items: center;
  justify-content: space-around;
  width: 100%;
  margin: 20px;
  div {
    width: 100%;
    padding: 20px;
    text-align: center;
  }

  @media (min-width: 600px) {
    width: 70%;
    margin: 20px auto;
  }

  @media (min-width: 900px) {
    width: 50%;
  }
`;

export const IconWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${colors.peach};
`;

export const ContentWrapper = styled.div`
  background: #f8f9f9;
  padding: 8px;
  p {
    margin-top: 0;
  }
`;

export const ErrorMessage = styled.p`
  font-size: 16px;
  font-weight: 600;
  margin-bottom: 4px;
  color: ${colors.charcoalGrey};
`;

export const ErrorHelpMessage = styled.p`
  font-size: 12px;
  color: ${colors.steel};
`;
