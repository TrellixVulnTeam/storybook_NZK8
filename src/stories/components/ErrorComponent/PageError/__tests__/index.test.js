import React from 'react';
import { mount } from 'enzyme';
import PageError from 'components/ErrorComponent/PageError/index';

describe('<PageError />', () => {
  it('should render correctly with mandatory props', () => {
    const sut = mount(<PageError name="Name" />);

    expect(sut).toMatchSnapshot();
  });
});
