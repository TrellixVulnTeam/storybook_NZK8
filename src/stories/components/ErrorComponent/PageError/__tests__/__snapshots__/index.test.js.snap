// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageError /> should render correctly with mandatory props 1`] = `
<PageError
  name="Name"
>
  <styled__WrapperDiv>
    <div
      className="styled__WrapperDiv-w1dqal-1 jaTVfF"
    >
      <styled__IconWrapper>
        <div
          className="styled__IconWrapper-w1dqal-2 haIZXu"
        >
          <styled__AlertIcon>
            <test-file-stub
              className="styled__AlertIcon-w1dqal-0 hVdZQz"
            />
          </styled__AlertIcon>
        </div>
      </styled__IconWrapper>
      <styled__ContentWrapper>
        <div
          className="styled__ContentWrapper-w1dqal-3 khjyeA"
        >
          <styled__ErrorMessage>
            <p
              className="styled__ErrorMessage-w1dqal-4 lolziL"
            >
              Something went wrong while loading
               Name
            </p>
          </styled__ErrorMessage>
          <styled__ErrorHelpMessage>
            <p
              className="styled__ErrorHelpMessage-w1dqal-5 eixZob"
            >
              Please try again in sometime
            </p>
          </styled__ErrorHelpMessage>
        </div>
      </styled__ContentWrapper>
    </div>
  </styled__WrapperDiv>
</PageError>
`;
